domain/upgrade/state/state.go

```diff
@@ -375,53 +349,98 @@ AND (
 	return nil
 }
 
-// ActiveUpgrade returns the uuid of the active upgrade.
-// The active upgrade is any upgrade that is not in the StepsCompleted state.
+// ActiveUpgrade returns the uuid of the active upgrade. The active upgrade is
+// any upgrade that is not in the StepsCompleted state. It returns a NotFound
+// error if there is no active upgrade.
 func (st *State) ActiveUpgrade(ctx context.Context) (domainupgrade.UUID, error) {
 	db, err := st.DB()
 	if err != nil {
 		return "", errors.Trace(err)
 	}
-	var activeUpgrade domainupgrade.UUID
-	q := "SELECT (uuid) FROM upgrade_info WHERE state_type_id < ?"
 
-	err = db.StdTxn(ctx, func(ctx context.Context, tx *sql.Tx) error {
-		row := tx.QueryRowContext(ctx, q, upgrade.StepsCompleted)
-		if err := row.Scan(&activeUpgrade); err != nil {
+	info := Info{
+		StateIDType: int(upgrade.StepsCompleted),
+	}
+
+	stmt, err := st.Prepare(`
+SELECT &Info.uuid
+FROM upgrade_info 
+WHERE state_type_id < $Info.state_type_id
+`, info)
```

Comment:
I don't mind if you ignore me here

But personally I prefer passing empty structs in to Prepare.

I think if you pass a specific instance of a struct, it communicates that it's specific value is important, which is obviously not the case with sqlair

An empty struct makes it clear this is not the case

Reply:
I agree, but Joe does not. So I've been doing it his way for consistency.

Reply:
Time for a mutiny?